department-of-veterans-affairs/caseflow
# Prernadev/appeals 36748

Resolves [Transmit Time Zone Information from "Schedule Veteran" Form](https://jira.devops.va.gov/browse/APPEALS-36748)

# Description
Value Statement
As a hearings coordinator I need all information relevant to the timing of a hearing sent from the forms in which I schedule hearings provided to the system so that it can accurately persist when a hearing is set to take place.

Notes
This context will be provided by the submission behavior of the [ScheduleVeteranForm](https://github.com/department-of-veterans-affairs/caseflow/blob/79e992e16b9296c0def4ab4be59e7819bf72f805/client/app/hearings/components/ScheduleVeteranForm.jsx) component. The form values are "packaged" [here](https://github.com/department-of-veterans-affairs/caseflow/blob/58878cdd8022747cdc6f5f288518975c1a38591c/client/app/hearings/components/ScheduleVeteran.jsx#L241-L248) in the parent ScheduleVeteran component before being sent to the backend.

The changes involved updating the front-end dropdown component values to pass the entire time string including the AM/PM and timezone context to the Ruby back-end on form submission.

## Acceptance Criteria
- [x] Whenever a hearings coordinator submits a "Schedule Veteran" form in order to schedule a hearing the value of the scheduled_time_string passed includes the name of the time zone of the regional office a user has selected for the hearing to be held at
Example of a current value: "13:00"
Example what a value should look like (assume St. Petersburg, FL is the RO in this example): "1:00 PM Eastern Time (US & Canada)"

## Testing Plan
###Test to verify hearing time
1. In Caseflow, try to Schedule a Veteran's hearing by selecting the "Schedule Veteran" button
2. Try to create a hearing for a location not based in EST
3. After selecting the appellant, select the "Schedule Veteran" option in the Actions dropdown. You will be directed to the "Schedule Appellant for a Hearing" page
4. On the "Schedule Apellant for a Hearing" page, right-click to inspect the element on the Hearing Time dropdown component. Select an option from the dropdown and the value should show the entire time string including the timezone context, similar to what is displayed in the screenshot.
5. Fill out the rest of the Schedule Veteran form and click the "Schedule" button
6. Go to the Daily Docket page for the hearing you've just scheduled and verify that the time being displayed is correct for the timezone.
![Screenshot 2024-01-18 at 5 00 26 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/132498915/202e7255-e142-4119-8581-85950312b24a)


###Test to verify hearing time in emails
1. If you're running this locally, make sure you have the mail dev server running
2. In Caseflow, try to Schedule a Veteran's hearing by selecting the "Schedule Veteran" button
3. Try to create a hearing for a location not based in EST
4. After selecting the appellant, select the "Schedule Veteran" option in the Actions dropdown. You will be directed to the "Schedule Appellant for a Hearing" page
5. On the "Schedule Apellant for a Hearing" page, change the meeting type to "Virtual" and fill out the apellant and PoA's emails
6. Fill out the rest of the Schedule Veteran form and click the "Schedule" button
8. Go to the Daily Docket page for the hearing you've just scheduled and verify that the time being displayed is correct for the timezone.
9. Check the mail dev server and verify that the sent emails are showing the correct time as per the selected timezone



# Best practices
## Code Documentation Updates
- [ x] Add or update code comments at the top of the class, module, and/or component.

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?
